src/components/InstrumentPicker/InstrumentList.tsx

```diff
@@ -31,51 +52,56 @@ class InstrumentList extends React.Component<
     };
   }
 
-  componentWillReceiveProps(args: any) {
-    this.setState({
-      data: args.instruments
-    });
+  componentWillReceiveProps(args: InstrumentListProps) {
+    this.setState(this.sortPickerInstruments(args.instruments));
   }
 
   componentDidMount() {
-    this.props.change();
-    const {
-      sortByParam,
-      direction,
-      state
-    } = this.props.getInstrumentPickerSortingParameters();
-    this.setState(
-      sortData(this.props.instruments, sortByParam, direction, state)
-    );
+    this.setState(this.sortPickerInstruments(this.props.instruments));
   }
 
-  sort = (sortByParam: string, sortDirection: string) => {
-    const direction =
-      this.state.sortByParam === sortByParam
-        ? sortDirection
-        : SortDirection.DESC;
-    const instrumentsToBeSorted = this.props.instruments.filter(
-      i => !!i[sortByParam]
-    );
-    const shouldNotBeSorted = this.props.instruments.filter(
-      i => !i[sortByParam]
-    );
+  sortPickerInstruments = (
+    instruments: InstrumentModel[],
+    state?: TableSortState,
+    param?: string,
+    sortDirection?: string
```

Comment:
It make sense to rename it to direction to use same style you use for param parameter